src/NCronJob/Configuration/Builder/DependencyBuilder.cs

```diff
@@ -3,21 +3,20 @@ namespace NCronJob;
 /// <summary>
 /// Represents the builder for the dependent jobs.
 /// </summary>
-public sealed class DependencyBuilder<TPrincipalJob>
-    where TPrincipalJob : IJob
+public sealed class DependencyBuilder
 {
     private readonly List<JobDefinition> dependentJobOptions = [];
 
     /// <summary>
     /// Adds a job that runs after the principal job has finished with a given <paramref name="parameter"/>.
     /// </summary>
     /// <remarks>
-    /// <typeparamref name="TJob"/> will automatically be registered in the container. There is no need to call <see cref="NCronJobOptionBuilder.AddJob{TJob}"/> for the dependent job.
+    /// <param name="jobType">The type of the job to be executed. This type must implement <see cref="IJob"/>.</param>
+    /// <param name="parameter">The parameter to be passed to the job when it is executed.</param>
     /// </remarks>
-    public DependencyBuilder<TPrincipalJob> RunJob<TJob>(object? parameter = null)
-        where TJob : IJob
+    public DependencyBuilder RunJob(Type jobType, object? parameter = null)
```

Comment:
I know we moved some of the generic version out as an extensions, but here I feel we open up the pit of failure.

When is there a need for the user has to dynamically configure that?

Reply:
In order to move forward with this PR, I've dropped that commit and parked it in a different branch.